mcdreforged/cli/cli_entry.py

```diff
@@ -157,3 +176,121 @@ def write(base_path: str, *, directory_only: bool):
+			return default if len(r) == 0 else r
+
+	if path is None:
+		path = ask('Plugin work directory', default='.')
+
+	metafile = os.path.join(path, plugin_constant.PLUGIN_META_FILE)
+	if os.path.exists(metafile):
+		writeln('[ERROR] Meta file "{}" already exists'.format(metafile))
+		sys.exit(1)
+
+	if pid is None:
+		pid = ask('Id', default=os.path.basename(os.path.abspath(path))).strip()
+	if name is None:
+		name = ask('Name', default=pid).strip()
+	if description is None:
+		description = ask('Description', default='This is a plugin for MCDR').strip()
+	if authors is None:
+		authors = ask("Author(s), split with ':'")
+	if link is None:
+		link = ask('Main page link', skip=True)
+	if resources is None:
+		resources = ask("Resource(s), split with ':'", skip=True)
+	if entrypoint is None:
+		entrypoint = ask('Entry point', skip=True)
+	if archive_name is None:
+		archive_name = ask('Archive name', skip=True)
+
+	metadata = {
+		'id': pid,
+		'version': '1.0.0',
+		'name': name,
+		'description': description,
+		'dependencies': {
+			'mcdreforged': '>=2.0.0'
+		}
+	}
+	if entrypoint is not None:
+		metadata['entrypoint'] = entrypoint
+	if authors is not None:
+		metadata['author'] = list(set((a.strip() for a in authors.split(':'))))
+	if link is not None:
+		metadata['link'] = link
+	if resources is not None:
+		metadata['resources'] = list(set(resources.split(':')))
+	if archive_name is not None:
+		metadata['archive_name'] = archive_name
+
+	with open(metafile, 'w') as fd:
+		json.dump(metadata, fd, indent=4)
+		writeln('Created meta file "{}"'.format(metafile))
+
+	with open(os.path.join(path, 'requirements.txt'), 'w') as fd:
+		fd.write('# Add your python package requirements here, just like regular requirements.txt\n')
+
+	entry, point = (pid, '__init__.py') if entrypoint is None else entrypoint.split('.', 1)
+	if not os.path.exists(os.path.join(path, entry)):
+		os.mkdir(os.path.join(path, entry))
+	entrypointf = os.path.join(path, entry, point)
+	with open(entrypointf, 'w') as fd:
+		fd.write('# Write your codes here\n')
+		writeln('Created entrypoint "{}"'.format(entrypointf))
+
+
+	status, _ = run_sh_cmd('git --version')
+	if status == 0:
+		if not os.path.exists(os.path.join(path, '.git')):
+			writeln('Initing git workspace...')
+			must_run_cmd('git -C {c} init -q'.format(c=path), quiet=quiet)
+		gitlink = ask("Input this plugin's git repository link", skip=True)
+		if gitlink is not None:
+			must_run_cmd('git -C {c} remote add origin {l}'.format(c=path, l=gitlink), quiet=quiet)
+			writeln('Commiting...')
+			must_run_cmd('git -C {c} add .'.format(c=path), quiet=quiet)
+			must_run_cmd("git -C {c} commit -q -m 'First commit'".format(c=path), quiet=quiet)
+			must_run_cmd('git -C {c} push -q -u origin master'.format(c=path), quiet=quiet)
+	else:
+		writeln('[WARN] Cannot find command `git`')
+
+def try_decodes(string, encodes):
```

Comment:
Hey, keep the same indentation style

Reply:
Oh sorry, I wasn't care about it

(In fact, I just copy-paste